### PR TITLE
Node js quickstart v2

### PR DESCRIPTION
**Pull request by prajwal-pai77**

### CODE REVIEW

File: __test__/api-airline-delete.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  connectToDatabase, // couchbase
+  app, // REST application
+} from './imports'
+
+afterAll(async () => {
+  const { cluster } = await connectToDatabase()
+  await cluster.close()
+})
+
+describe('DELETE /api/v1/airline/{id}', () => {
```
Comment (nithishr) [inline __test__/api-airline-delete.test.js:15]: Shouldn't there be a test that deletes a non-existing document?

File: __test__/api-airport-list.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        })
+        .catch((e) => console.log(`test airport2 insert failed: ${e.message}`))
+    })
+    test('should respond with status code 200 OK and return the documents', async () => {
+      const response = await request(app).get(`/api/v1/airport/list`).query({
+        offset: 0,
+        limit: 5,
+        country: 'Test Country',
+      })
+      expect(response.statusCode).toBe(200)
```
Comment (nithishr) [inline __test__/api-airport-list.test.js:55]: Can you also check if the documents have the same country?

File: __test__/imports.js [hunk 3]
```diff
@@ -8,8 +8,12 @@ import { app } from '../src/app'
...
 
 module.exports = {
-  request, describe, test, expect, //supertes
-  bcrypt, v4,                      // utilities
-  connectToDatabase,      // couchbase
-  app                              // REST application
+  request,
+  describe,
+  test,
+  expect, //supertes
```
Comment (nithishr) [inline __test__/imports.js:14]: What is the supertes? Is it supertest?

File: config/dev.env [hunk 4]
```diff
@@ -1,7 +1,7 @@
-CB_URL=localhost
-CB_USER=Administrator
-CB_PASS=password
-CB_BUCKET=user_profile
-IS_CAPELLA=false
+DB_CONN_STR=couchbases://<identifier>.cloud.couchbase.com
+DB_USERNAME=
+DB_PASSWORD=
+DB_BUCKET_NAME=travel-sample
+IS_CAPELLA=true
```
Comment (nithishr) [inline config/dev.env:5]: Do you need the IS_CAPELLA anymore? 
Wan Profile will work with both on prem & capella. It is only setting a few timeouts.

File: config/dev.env [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
 APP_PORT=3000
-DELAY=5000
+DELAY=5000
```
Comment (nithishr) [inline config/dev.env:7]: Is DELAY getting used? I would expect it to have been part of the older index & bucket creation logic.

**Review by nithishr — CHANGES_REQUESTED** [review]
Some minor improvements:
Testing
 - Can you add tests for validation of the fields?
 - Negative (404) tests
 - Some additional checks in the responses (values, length, country, etc) in some cases

Double check the environment variables

File: README.md [hunk 6]
```diff
@@ -1,66 +1,166 @@
...
 ## Prerequisites
 
 To run this prebuilt project, you will need:
 
-- A Couchbase Capella cluster or Couchbase 7 installed locally
-- NodeJS & NPM (v12+)
-- Code Editor
+- [Couchbase Capella](https://www.couchbase.com/products/capella/) cluster with [travel-sample](https://docs.couchbase.com/python-sdk/current/ref/travel-app-data-model.html) bucket loaded.
+  - To run this tutorial using a self managed Couchbase cluster, please refer to the [appendix](#running-self-managed-couchbase-cluster).
+- [Node js](https://nodejs.org/en/download) installed on local machine.
```
Comment (nithishr) [inline README.md:22]: Can you also add the minimum version of Node that is supported?

File: __test__/api-airline-delete.test.js (hunk 1 above)
Comment (prajwal-pai77) [reply __test__/api-airline-delete.test.js:15]: Added 404 tests which deletes invalid doc id

File: __test__/api-airport-list.test.js (hunk 2 above)
Comment (prajwal-pai77) [reply __test__/api-airport-list.test.js:55]: Added additional checks for all routes

File: __test__/imports.js (hunk 3 above)
Comment (prajwal-pai77) [reply __test__/imports.js]: Thanks, renamed to supertest

File: config/dev.env (hunk 4 above)
Comment (prajwal-pai77) [reply config/dev.env:5]: You are right, it is not required, removed the IS_CAPELLA flag.

File: config/dev.env (hunk 5 above)
Comment (prajwal-pai77) [reply config/dev.env:7]: Removed DELAY and the corresponding delay.js

File: README.md (hunk 6 above)
Comment (prajwal-pai77) [reply README.md]: Added the minimum version of node based on the couchbase sdk package.json configuration.

File: README.md [hunk 7]
```diff
@@ -1,66 +1,166 @@
...
 ## Prerequisites
 
 To run this prebuilt project, you will need:
 
-- A Couchbase Capella cluster or Couchbase 7 installed locally
-- NodeJS & NPM (v12+)
-- Code Editor
+- [Couchbase Capella](https://www.couchbase.com/products/capella/) cluster with [travel-sample](https://docs.couchbase.com/python-sdk/current/ref/travel-app-data-model.html) bucket loaded.
+  - To run this tutorial using a self managed Couchbase cluster, please refer to the [appendix](#running-self-managed-couchbase-cluster).
+- [Node js v16+](https://nodejs.org/en/download) installed on local machine.
```
Comment (nithishr) [inline README.md:22]: I think we should put the Node Version as any LTS version like the SDK does. Coz Node 16 is already EOL. 
https://docs.couchbase.com/nodejs-sdk/current/project-docs/compatibility.html

**Review by nithishr — APPROVED** [review]
Looks good. Can you please update the Node version & merge?

File: README.md (hunk 7 above)
Comment (prajwal-pai77) [reply README.md]: Updated to be Node js LTS version.

**Pull request merged**